crypto/bls/signature_batch.go

```diff
@@ -82,6 +145,10 @@ func (s *SignatureBatch) RemoveDuplicates() (int, *SignatureBatch, error) {
 	sigs := s.Signatures[:0]
 	pubs := s.PublicKeys[:0]
 	msgs := s.Messages[:0]
+	var descs []string
+	if s.Descriptions != nil {
```

Comment:
same thing here, do not have a nil check here. The correct thing to do would be to make sure all signature sets always have an attached description.